### DISCUSSION TITLE
Create signed url to upload storage object and track progress

### DISCUSSION THREAD

**Discussion by kk21** [supabase/supabase#4034] [Feature Requests]
Hi,

Is it possible to provide signed url to upload storage object?

This may allow web clients to track the upload progress of large files using axios / XMLHttpRequest.

Thanks.

**Comment by bruceharrison1984** [comment]
I'm interested in this as well. I'm was going to use react-uploady for my image uploads, but it only supports URL destinations.

**Comment by giovanniincammicia** [comment]
Is there any plan to support this? I think it would be very useful for Background Fetch in PWAs. This way we'd be able to upload files in background and have a nice native-like progress experience.

**Comment by inian** [comment]
You can listen to progress events with our resumable uploads feature in [Storage v3](https://supabase.com/blog/storage-v3-resumable-uploads). Here are the docs on [resumable uploads](https://supabase.com/docs/guides/storage/uploads#resumable-upload).

You can create signed urls for upload too - [docs](https://supabase.com/docs/reference/javascript/storage-from-uploadtosignedurl).
